engine/extensions-xpath/source/org/pentaho/reporting/engine/classic/extensions/datasources/xpath/LegacyXPathTableModel.java

```diff
@@ -96,12 +104,20 @@ public LegacyXPathTableModel( final ResourceData xmlResource,
       columnTypes = new ArrayList<Class>();
       columnNames = new ArrayList<String>();
 
+      DocumentBuilderFactory df = DocumentBuilderFactory.newInstance();
+      final Configuration configuration = ClassicEngineBoot.getInstance().getGlobalConfig();
+      if ( !"true".equals( configuration
+              .getConfigProperty( "org.pentaho.reporting.designer.support.dtds.enable" ) ) ) {
```

Comment:
Please change the property name to match the package of the XPAth datasource. It is not just report-designer but also BI-Server that are affected.

Use "org.pentaho.reporting.engine.classic.extensions.datasources.xpath.EnableDTDs" as property name, for instance.